images/egress/dns-proxy/egress-dns-proxy.sh

```diff
@@ -170,6 +202,18 @@ function run() {
  }
 
 if [[ "${EGRESS_DNS_PROXY_MODE:-}" == "unit-test" ]]; then
+
+    declare -A DNS_SERVERS=()
+    if [[ "${EGRESS_DNS_SERVERS+isset}" ]]; then
+        servers=(${EGRESS_DNS_SERVERS//;/ })
+        for server in "${servers[@]}"; do
+            si=(${server//:/ })
+            if [[ "${#si[@]}" -eq "2" ]]; then
+                DNS_SERVERS[${si[0]}]=${si[1]}
+            fi
```

Comment:
Is there a valid case where `[[ ${#si[@]} -ne 2 ]]`? If not, why not print an error message and exit immediately?

Reply:
done